core/templates/pages/exploration-editor-page/modal-templates/metadata-version-history-modal.component.ts

```diff
@@ -0,0 +1,225 @@
+// Copyright 2022 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for state changes modal.
+ */
+
+import { Input, OnInit } from '@angular/core';
+import { Component } from '@angular/core';
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+import { ConfirmOrCancelModal } from 'components/common-layout-directives/common-elements/confirm-or-cancel-modal.component';
+import { ExplorationMetadata } from 'domain/exploration/ExplorationMetadataObjectFactory';
+import { ContextService } from 'services/context.service';
+import { HistoryTabYamlConversionService } from '../services/history-tab-yaml-conversion.service';
+import { VersionHistoryBackendApiService } from '../services/version-history-backend-api.service';
+import { VersionHistoryService } from '../services/version-history.service';
+
+interface headersAndYamlStrs {
+  leftPane: string;
+  rightPane: string;
+}
+
+interface mergeviewOptions {
+  lineNumbers: boolean;
+  readOnly: boolean;
+  mode: string;
+  viewportMargin: number;
+}
+
+@Component({
+  selector: 'oppia-metadata-version-history',
+  templateUrl: './metadata-version-history-modal.component.html',
+})
+export class MetadataVersionHistoryModalComponent
+  extends ConfirmOrCancelModal implements OnInit {
+  @Input() committerUsername!: string;
+  @Input() oldVersion: number | null = null;
+  @Input() newMetadata!: ExplorationMetadata;
+  @Input() oldMetadata!: ExplorationMetadata;
+  yamlStrs: headersAndYamlStrs = {
```

Comment:
Instead of the terminology "leftPane" and "rightPane", should we use a name that explains what the pane represents? It is possible that we change the order of the panes in the display but keep the original behaviour -- which can result in a disconnect between the component controller and the template.

Reply:
done.